dso/model.py

```diff
@@ -380,6 +397,16 @@ def key(self) -> str:
         return self.finding.key
 
 
+@dataclasses.dataclass(frozen=True)
+class SastFinding(Finding):
+    sast_statuses: list[SastStatus] | SastStatus
+    sub_type: SastSubType
+
+    @property
+    def key(self) -> str:
+        return _as_key(self.sast_statuses, self.sub_type)
```

Comment:
In case `sast_statuses` is of type `list[SastStatus]`, you'll have to take care of sorting it first (to keep the key stable) + unpack the list so that it can be processed by `_as_key`.
On another note: Do we actually still require `sast_status(es)` to allow lists? The only status currently is `no-linter`, and if we'd like to add something like `has-linting-findings` in the future, it is logically not possible that both of them occur at the same time.